internal/controller/object/object.go

```diff
@@ -282,11 +302,17 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{}, err
 	}
 
-	var last *unstructured.Unstructured
-	if last, err = getLastApplied(cr, observed); err != nil {
+	// observation contains the extracted state of the observed object that
+	// should be compared with the desired state of the object manifest to
+	// decide whether the object is up-to-date or not.
+	// If serverSideApply is enabled, we will extract the state from the
+	// observed object, otherwise we will extract the state from the last
+	// applied annotation.
```

Comment:
who owns defaults? How safe is a comparison here? Don't know the logic enough of the provider. What is this value used for?

Reply:
which defaults do you mean? 

Basically, in Observe, we are trying to decide whether the resource is up to date or not. Prior to SSA, we were using "kubectl.kubernetes.io/last-applied-configuration" to see which manifest was applied and if it is different than the current one we were deciding that we need an update. Obviously, this has a good amount of downsides (see the issues closed with this PR) and now we are trying to do the same with SSA in a more proper way. We are using unstructured extractor to extract the observed state and will use it to make the comparison. However, currently SSA is not working as advertised due to issues [here](https://github.com/crossplane-contrib/provider-kubernetes/pull/246#discussion_r1606334812) and I had to use a workaround with dry run to extract the observed state.

Reply:
Same question as discussed in https://github.com/crossplane-contrib/provider-kubernetes/pull/246#discussion_r1605251719.